### PR TITLE
Core: Allow C and JSON keymaps to be used together

### PR DESCRIPTION
**Pull request by Erovia**
## Description

Keep your keymaps nice and simple with JSON keymaps, but use the full
power of macros written in C.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by zvecr — CHANGES_REQUESTED** [review]
Honestly, I would prefer a lighter touch such as what #13197 provided.

File: quantum/quantum.c [hunk 1]
```diff
@@ -108,6 +108,10 @@ __attribute__((weak)) bool process_action_kb(keyrecord_t *record) {
 }
 
 __attribute__((weak)) bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    return process_record_json(keycode, record);
```
Comment (zvecr) [inline quantum/quantum.c:111]: Really not a fan of introducing this convention, its already bad enough that keyboards forget to call `process_record_user`. Now there will be a new convention, and a bunch of "broken" keyboards as they do `process_record_kb -> process_record_user`.

**Review by zvecr — CHANGES_REQUESTED** [review]
Honestly, I would prefer a lighter touch such as what https://github.com/qmk/qmk_firmware/pull/13197 provided.

File: quantum/quantum.c (hunk 1 above)
Comment (Erovia) [reply quantum/quantum.c:111]: I'm not married to this implementation, but other approaches seemed more "hacky".
Although keyboard's `process_record_kb` can certainly brake this, keyboards breaking convention might be an other problem we have to tackle in some other way.

I marked this PR to be Draft until we decide how to proceed.

Comment (drashna) [reply quantum/quantum.c:111]: If anything, I would say that it should be moved into to the rest of the process_record_* handling is done, just like a normal feature.

**Comment by zvecr** [comment]
Closing for now as #19939 was added.

**Pull request closed without merging**